docs/timezone-draft.md

```diff
@@ -0,0 +1,169 @@
+
+Temporal.TimeZone = FrozenTimeZoneImplementation;
+```
+
+## Implementation of a custom time zone
+
+### TimeZone interface
+
+```javascript
+class Temporal.TimeZone {
+  /** Sets the [[Identifier]] internal slot to @id, and creates the
+   * [[InitializedTemporalTimeZone]] internal slot */
+  constructor(id : string) : Temporal.TimeZone;
+
+  // Methods that a subclassed custom time zone must implement
+
+  /** Given an instant (in epoch nanoseconds) returns this time zone's
+   * corresponding UTC offset, in nanoseconds (signed). */
+  getOffsetAtInstant(epochNs : bigint) : bigint;
+
+  /** Given the calendar/wall-clock time, returns an array of 0, 1, or
+   * 2 instants (in epoch nanoseconds) that are possible points on the
+   * timeline corresponding to it. */
+  possibleInstants(
+    isoYear : number,
+    isoMonth : number,
+    isoDay : number,
+    hour : number,
+    minute : number,
+    second : number,
+    millisecond : number,
+    microsecond : number,
+    nanosecond : number
+  ) : array<bigint>;
+
+  *getTransitions(startingPoint : Temporal.Absolute) : iterator<Temporal.Absolute>;
+
+  // API methods that a subclassed custom time zone doesn't need to touch
+
+  get name() : string;
+  getDateTimeFor(absolute : Temporal.Absolute) : Temporal.DateTime;
+  getAbsoluteFor(
+      dateTime : Temporal.DateTime,
+      options?: object
+  ) : Temporal.Absolute;
+  getOffsetFor(absolute : Temporal.Absolute) : string;
+  toString() : string;
+  toJSON() : string;
+
+  static from(item : any, options?: object) : Temporal.TimeZone;
+  static idToTimeZone(id: string) : Temporal.TimeZone;
+  static [Symbol.iterator]() : iterator<Temporal.TimeZone>;
+}
+```
+
+The `getOffsetAtInstant()` and `possibleInstants()` methods are conceptually "private", meaning that they are not intended as API to be normally used in user code.
+Users will not encounter them unless they are implementing a custom time zone.
+
+All the other methods are implemented in terms of these two and the value of the _[[Identifier]]_ internal slot.
+For example, `getOffsetFor()` and `getDateTimeFor()` call `getOffsetAtInstant()`, and `getAbsoluteFor()` calls both.
+
+> **FIXME:** These methods are named "instant" and not "absolute" to reflect that they don't take a `Temporal.Absolute` object.
+> Help is welcome in determining the color of this bike shed.
+
+Alternatively, a custom time zone doesn't have to be a subclass of `Temporal.TimeZone`.
+In this case, it can be a plain object, which must implement all of the non-static methods in the interface, except for `getOffsetAtInstant()` and `possibleInstants()`.
+
+> **FIXME:** This means we have to remove any checks for the _[[InitializedTemporalTimeZone]]_ slot in any APIs outside of `Temporal.TimeZone`.
+
+> **FIXME:** `Temporal.TimeZone` is supposed to be an iterable through all time zones known to the implementation, but what do we do about custom time zones there?
```

Comment:
We should decide how to do this in #294 for calendars and apply the same solution here.